src/scripts.c

```diff
@@ -51,7 +51,11 @@ static void update_triggers(void)
 {
 	__attribute__((unused)) int ret = 0;
 
-	ret = system("/usr/bin/systemctl --no-block daemon-reload");
+	if (need_systemd_reexec) {
+		ret = system("/usr/bin/systemctl daemon-reexec");
+	}
```

Comment:
the reload must not have --no-block, because the execution of the triggers really needs to be the NEW triggers, not the old version triggers.....